### PR TITLE
make the code easy to understand

### PR DESCRIPTION
**Pull request by divinerapier**
Hi, I find the code is difficult to understand. So I make the pull request.

Thanks.

### CODE REVIEW

**Comment by Thor-wl** [comment]
Welcome! If find any other code hard to understand, don't hesitate to optimize or make comments.

File: pkg/scheduler/api/resource_info.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func NewResource(rl v1.ResourceList) *Resource {
 
 // IsEmpty returns bool after checking any of resource is less than min possible value
 func (r *Resource) IsEmpty() bool {
-	if !(r.MilliCPU < minMilliCPU && r.Memory < minMemory) {
+	if r.MilliCPU >= minMilliCPU || r.Memory >= minMemory {
 		return false
```
Comment (william-wang) [inline pkg/scheduler/api/resource_info.go:98]: The minMilliCPU and mini minMemory is float64 type. Do not use equal to compare.

Comment (divinerapier) [reply pkg/scheduler/api/resource_info.go:98]: Thanks for your advice. I have two ideas to deal with the problem. PTAL @william-wang , thanks a lot.

The first:
``` go
func (r *Resource) IsEmpty() bool {
	// Do not use equal to compare floats.
	const precision = 1e-6
	if r.MilliCPU > (minMilliCPU-precision) || r.Memory > (minMemory-precision) {
		return false
	}

	for _, rQuant := range r.ScalarResources {
		if rQuant >= minMilliScalarResources {
			return false
		}
	}

	return true
}
```
The second:
``` go
func (r *Resource) IsEmpty() bool {
	// Do not use equal to compare floats. Compare if a is greater than or equal to b
	geFn := func(a, b float64) bool {
		const precision = 1e-6
		return a > (b - precision)
	}
	if geFn(r.MilliCPU, minMilliCPU) || geFn(r.Memory, minMemory) {
		return false
	}

	for _, rQuant := range r.ScalarResources {
		if rQuant >= minMilliScalarResources {
			return false
		}
	}

	return true
}
```

**Comment by k82cn** [comment]
when will we replace those helper func with Rat :)

**Comment by divinerapier** [comment]
@william-wang  Thanks for your advice. I updated the code, please take a look, thanks a lot.

**Comment by william-wang** [comment]
> when will we replace those helper func with Rat :)

@Thor-wl is working on the helper replacement with RAT. It will be ready in 1.1.1 version on January 2021.

**Comment by Thor-wl** [comment]
Accuracy development is on the way.

**Pull request closed without merging**